Userland/Libraries/LibCompress/PackBitsDecoder.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2023, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "PackBitsDecoder.h"
+
+namespace Compress::PackBits {
+
+ErrorOr<ByteBuffer> decode_all(ReadonlyBytes bytes, Optional<u64> expected_output_size)
+{
+    auto memory_stream = make<FixedMemoryStream>(bytes);
+
+    ByteBuffer decoded_bytes;
+    TRY(decoded_bytes.try_ensure_capacity(expected_output_size.value_or(0)));
+
+    while (memory_stream->remaining() > 0 && decoded_bytes.size() <= expected_output_size.value_or(NumericLimits<u64>::max())) {
+        auto const length = TRY(memory_stream->read_value<u8>());
```

Comment:
Once we know this, we could probably `try_ensure_capacity` for an entire chunk, and then use `append_unchecked` when actually appending.

Reply:
Let's wait until this is a bottleneck to do it